### DISCUSSION TITLE
postUpgradeTasks not triggered at all

### DISCUSSION THREAD

**Discussion by FelixJacobi** [renovatebot/renovate#19584] [Request Help] [answered]
Hi everyone,

I am trying to deal with a legacy PHP application which have composer's vendor checked-in in GIT and is depended on `composer install` running from a specific location. What I want to archive:

* Running `composer install`  in that specific location
* Committing vendor directory after that to GIT

I've written a small shell script for that and put that as `/usr/local/bin/fix-v3-vendor` in a custom renovate docker image extended from the official one:

~~~shell
#!/bin/sh
set -e

ORIG_DIR="$PWD"
WEB_DIR="/usr/share/iserv/web"
VENDOR_DIR="$WEB_DIR/vendor"
mkdir -p -v "$WEB_DIR"
rm -f "$VENDOR_DIR"
ln -s "$ORIG_DIR/web/vendor" "$VENDOR_DIR"
composer install --no-ansi --no-interaction --no-scripts --no-autoloader --no-plugins --ignore-platform-req='ext-*' --ignore-platform-req='lib-*' -d "$WEB_DIR"
git add web/vendor

~~~

Next, I allowed that command in the global configuration (exposed  to Renovate as `RENOVATE_CONFIG_FILE` using POSIX environment):

~~~json
{
  "allowPostUpgradeCommandTemplating": true,
  "allowedPostUpgradeCommands": ["^fix-v3-vendor$"]
}
~~~

Finally, I added `postUpgradeTasks` (while debugging, I tried per packageGroup and globally) to the `renovate.json`:

~~~json
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "config:base",
    ":disableDependencyDashboard"
  ],
  "enabledManagers": ["composer"],
  "rangeStrategy": "update-lockfile",
  "rebaseWhen": "conflicted",
  "major": { "enabled": false },
  "prHourlyLimit": 0,
  "prConcurrentLimit": 0,
  "ignorePaths": ["web/vendor/**", "web/node_modules/**"],
  "packageRules": [
    {
      "matchPackagePrefixes": ["horde/"],
      "groupName": "horde packages",
      "groupSlug": "horde",
      "postUpgradeTasks": {
        "commands": ["fix-v3-vendor"],
        "fileFilters": ["web/vendor/**", "web/composer.json", "web/composer.lock"],
        "executionMode": "branch"
      }
    },
    {
      "matchPackagePrefixes": ["iserv/"],
      "groupName": "iserv packages",
      "groupSlug": "iserv",
      "postUpgradeTasks": {
        "commands": ["fix-v3-vendor"],
        "fileFilters": ["web/vendor/**", "web/composer.json", "web/composer.lock"],
        "executionMode": "branch"
      }
    },
    {
      "matchPackagePrefixes": ["psr/"],
      "groupName": "psr packages",
      "groupSlug": "psr",
      "postUpgradeTasks": {
        "commands": ["fix-v3-vendor"],
        "fileFilters": ["web/vendor/**", "web/composer.json", "web/composer.lock"],
        "executionMode": "branch"
      }
    },
    {
      "matchPackagePrefixes": ["sabre/"],
      "groupName": "sabre packages",
      "groupSlug": "sabre",
      "postUpgradeTasks": {
        "commands": ["fix-v3-vendor"],
        "fileFilters": ["web/vendor/**", "web/composer.json", "web/composer.lock"],
        "executionMode": "branch"
      }
    },
    {
      "matchPackagePrefixes": ["doctrine/"],
      "groupName": "doctrine packages",
      "groupSlug": "doctrine",
      "postUpgradeTasks": {
        "commands": ["fix-v3-vendor"],
        "fileFilters": ["web/vendor/**", "web/composer.json", "web/composer.lock"],
        "executionMode": "branch"
      }
    }
  ],
  "postUpgradeTasks": {
    "commands": ["fix-v3-vendor"],
    "fileFilters": ["web/vendor/**", "web/composer.json", "web/composer.lock"],
    "executionMode": "branch"
  }
}
~~~

But: The postUpgradeTasks are never executed. I also used `--log-file-level  debug` and `--log-file`, but there are never mentioned in the log at all. What I've missed?

We're using a self-hosted Renovate along with GitLab CI.

**Comment by rarkins** [comment, marked as answer]
Do you see the string `allowedPostUpgradeCommands` in your logs when Renovate starts up? Ie as part of config logging

**Reply by FelixJacobi** [reply]
> Do you see the string `allowedPostUpgradeCommands` in your logs when Renovate starts up? Ie as part of config logging 

Yes. It is visible. But I solved my case: Renovate consumes the `renovate.json` from the default branch of the GitLab rather than the one where CI runs and the config was modified (we have an internal review process which works with branches and Merge requests), this confused me a bit 😅 . So nothing wrong with Renovate, I guess I need to dive deeper into how it actually works.
